### ISSUE TITLE
Bootstrap 4

### ISSUE THREAD

**Issue by lgs** [todc/todc-bootstrap#172]
Bootstrap 4, anyone?

**Comment by acmetech** [comment]
Hi @lgs, I plan on supporting Bootstrap v4 when it reaches the release candidate stage.

**Comment by AugustDev** [comment]
Bootstrap 4 is in a great stage right now, it would be amazing if someone would make TODC for Bootstrap 4.

**Comment by acmetech** [comment]
@AugustDev, I've been keeping an eye on the progress being made with BS 4 and I don't feel that it is stable enough yet. BS 4 is scheduled to have a third beta. This means that they may still make significant changes.

FYI - I do have a local v4-dev branch of todc-bootstrap where I've been testing things out.

![todc bootstrap google-styled theme for bootstrap](https://user-images.githubusercontent.com/134294/33663670-e1fbc2ec-da5e-11e7-80b3-60d1d212fa94.png)

**Comment by AugustDev** [comment]
@acmetech thank you for reply, looks awesome! This is an awesome project and thank you for working on it.

**Comment by acmetech** [comment]
@AugustDev, you're welcome and thank you

**Comment by microtheta** [comment]
Bootstrap v4.0.0 released now...

**Comment by acmetech** [comment]
@microtheta, thank you

**Comment by chez14** [comment]
Is there any news now? I wanted to help.

**Comment by acmetech** [comment]
@chez14, Updates are coming...

**Comment by acmetech** [comment]
http://todc.github.io/todc-bootstrap/ has been updated with the latest updates. Please have a look.

I haven't uploaded the v4-dev branch yet...

**Comment by chez14** [comment]
Hmm are we still on the early stage now? The legendary-cute-looking button are not the Google's one :cry:...

**Comment by acmetech** [comment]
@chez14, what "legendary cute looking buttons" are you referring to? Please provide a link and specific feedback.

**Comment by chez14** [comment]
Oh sorry, i meant this

![image](https://user-images.githubusercontent.com/5555026/37503523-7ff48c0e-290b-11e8-9c74-b4bb49132ea1.png)

i was looking that button style, but i couldn't found any repo that more active than this.

And here's from the TODC's doc page for button.

![image](https://user-images.githubusercontent.com/5555026/37503564-bd86a84a-290b-11e8-9d27-0c310efd66c3.png)

It still has the default material-looking button. Am i in wrong direction here?

Thanks!

**Comment by acmetech** [comment]
@chez14, most of Google's web UIs have been updated to Material Design. Gmail is a notable exception.

If anyone else is interested, please let me know.

If I do add a Gmail style button, should it be the default style?
  - My thoughts: Yes. Since the default inputs, checkbox, radio, etc are still pretty close to the Gmail style.

**Comment by weiFans** [comment]
@acmetech Amazing Bootstrap 4, great work, when does it open to download?

**Comment by acmetech** [comment]
@weiFans, Thank you. I have a few more issues/updates I'd like to complete before uploading the alpha source. As far as the release version, not sure. Sorry, but when it's ready is as specific as I can be at this time.

**Comment by weiFans** [comment]
@acmetech Thank you for your reply. Don't worry, work hard, look forward to your work.

**Comment by weiFans** [comment]
@acmetech Just searched to [MDB](https://mdbootstrap.com/material-design-for-bootstrap/) ,Based on Material Design. 
Don't think Material Design is good to web practice. 
Material Design prefers Android, similar to IOS solutions. 
I think that this version of TODC Bootstrap v3.3.7 is great, only need to upgrade to Bootstrap v4.0.0.
Is your plan Material Design ?

**Comment by acmetech** [comment]
@weiFans, as stated in the README:

> This is a Google-styled theme for Bootstrap. Because I'm a fan of the new Google UI seen in Gmail, Docs, Calendar, etc

Since Calendar, Drive, Contacts, Maps, Photos, Search, etc are all based off of Material Design then that is the direction we are headed. The most obvious Google app that doesn't use Material Design is Gmail. In my opinion, the Gmail UI is looking a little dated compared to the UI of the other Google apps. v4 of TODC Bootstrap should still be able to give a UI similar to Gmail, but also Calendar. 

We are not trying to create a Material Design web framework/toolkit, we are tying to create a Google theme for Bootstrap. As also stated in the README:

> The project's goal isn't UI parity with Google Apps -- it's feature parity with Bootstrap, themed with Google's UI in mind.

Thanks for your feedback and please let me know if you have additional questions/concerns.

**Comment by AugustDev** [comment]
Download links at http://todc.github.io/todc-bootstrap/ do not work (both top-right corner and download section)

**Comment by acmetech** [comment]
@AugustDev that is correct. The v4.x documentation is currently only a preview with no available download at this time.
